clients/src/main/java/org/apache/kafka/clients/producer/Partitioner.java

```diff
@@ -45,4 +45,7 @@ public interface Partitioner extends Configurable, Closeable {
      * This is called when partitioner is closed.
      */
     void close();
+
+    default void onNewBatch(String topic, Cluster cluster, int prevPartition) {
```

Comment:
please consider using "casting" to invoke `RoundRobinPartitioner#onNewBatch` if needs